packages/d3fc-webgl/test/shaders/line/shaderSpec.js

```diff
@@ -0,0 +1,39 @@
+import * as fc from '../../../index';
+import { expectVertexShader, getShaders } from '../../util/vertextShaderUtil';
+
+it('test', () => {
+    const line = fc
+        .glLine()
+        .xScale(fc.glScaleLinear())
+        .yScale(fc.glScaleLinear());
+
+    const shaders = getShaders(line);
+
+    const attributes = {
+        aCorner: [-1, 0, 0],
+        aXValue: 2,
+        aYValue: 4,
+        aNextXValue: 3,
+        aNextYValue: 5,
+        aPrevXValue: 1,
+        aPrevYValue: 5,
+        aPrevPrevXValue: 0,
+        aPrevPrevYValue: 4,
+        aDefined: 1
+    };
+
+    const uniforms = {
+        uScreen: [500, 250],
+        uLineWidth: 2,
+        linear0Offset: [0, 0, 0, 0],
+        linear0Scale: [1, 1, 1, 1],
+        linear1Offset: [0, 0, 0, 0],
+        linear1Scale: [1, 1, 1, 1]
+    };
+
+    expectVertexShader(
+        shaders.vertexShader(),
+        attributes,
+        uniforms
+    ).toHaveGlPosition([1, 4.991055728090001, 0, 1]);
```

Comment:
@jamesmporter Can you make another method which takes a rounded input value to 4dp?

Reply:
As we're always dealing with floats, does it make sense to modify `toHaveGlPosition` to use `toBeCloseTo` or equivalent?

Reply:
Addressed in: https://github.com/d3fc/d3fc/pull/1422

Reply:
@chrisprice I that it's better to use the map/round solution than `toBeCloseTo` as the former gives us error information about the whole `gl_Position`.

```
shaderOutput.gl_Position = shaderOutput.gl_Position.map(
    x => Math.round((x + Number.EPSILON) * 10000) / 10000
);

return expect(shaderOutput).toEqual({
    gl_Position: glPosition
});
```

```
shaderOutput.gl_Position.forEach((x, i) => {
    expect(x + Number.EPSILON).toBeCloseTo(glPosition[i], 4);
});
```

Reply:
However, I've made the 4dp function the default (`toHaveGlPosition()`) and added another (`toHaveGlPositionExact()`), which compares the non-rounded float value.